src/puppetlabs/jdbc_util/pglogical.clj

```diff
@@ -123,11 +123,12 @@
   or :none if replication-mode is :none. Valid status values vary depending on
   thew role; see other replication-status functions for more information."
   [db replication-mode]
-  (case replication-mode
-    :source (provider-replication-status db)
-    :replica (replica-replication-status db)
-    :none :none
-    nil :none))
+  (with-timeout 4 :unknown
```

Comment:
https://github.com/puppetlabs/jdbc-util/blob/master/src/puppetlabs/jdbc_util/core.clj#L37 I copied it from the db-up? status check